UIKitTests/Gestures/TouchHandlingTests.swift

```diff
@@ -40,13 +40,27 @@ class TouchHandlingTests: XCTestCase {
         XCTAssertTrue(recognizer.onActionWasCalled)
     }
 
-    func testCancelsTouchesInView() {
+    func testCancelsTouchesInViewWithTapGestureRecognizer() {
+        let tapRecognizer = UITapGestureRecognizer()
+        tapRecognizer.cancelsTouchesInView = true
+        view.addGestureRecognizer(tapRecognizer)
+
+        handleTouchDown(CGPoint(x: 10, y: 10))
+        handleTouchUp(CGPoint(x: 10, y: 10))
+
+        XCTAssertFalse(view.touchesBeganWasCalled)
+        XCTAssertFalse(view.touchesEndedWasCalled)
+    }
+
+    func testCancelsTouchesInViewWithPanGestureRecognizer() {
         recognizer.cancelsTouchesInView = true
 
         handleTouchDown(CGPoint(x: 10, y: 10))
         handleTouchMove(CGPoint(x: 15, y: 10))
         handleTouchUp(CGPoint(x: 15, y: 10))
 
+        XCTAssertTrue(view.touchesBeganWasCalled)
```

Comment:
it would be so useful to construct UI Tests out of these for iOS – is that possible?

Reply:
unfortunately no as far as I know, we import UIKit with `@testable import` because we use internal API of UITouch

Reply:
we use the internal API in order to simulate touches programatically, but AFAIK UI Tests can also do that (in a more round-about way). Maybe there's a way of abstracting the simulated touches we do, that on iOS uses the UI Testing APIs and in ours just simulates touches directly?

Reply:
for sure – more wanted to share a general direction we might be able to take